### PR TITLE
Added snippets for Java unit tests

### PR DESCRIPTION
**Pull request by wesleyegberto**
Hi,

I'm adding the snippets to help write unit tests in Java.

I've created a new folder for Java, maybe could move the file `java.json` to it.

### CODE REVIEW

**Comment by OkelleyDevelopment** [comment]
Hey @wesleyegberto, my apologies on the delay reviewing this PR. 

I also wanted to let you know I will review this asap in the next couple days or so (and that I hadn't forgotten!) :sunglasses:

File: package.json [hunk 1]
```diff
@@ -157,6 +157,10 @@
         "language": "java",
         "path": "./snippets/java.json"
       },
+      {
+        "language": "java",
```
Comment (OkelleyDevelopment) [inline package.json:161]: I'm a big believer in "opt-in" on snippets. Perhaps you could change the language to be `java-testing`. 

This accomplishes two things: 
- Doesn't overload the users snippets if they don't want these particular testing snippets
- Users could then extend the snippets similar to this: 
```lua
require("luasnip").filetype_extend("java", { "java-testing" })
```

**Review by OkelleyDevelopment — CHANGES_REQUESTED** [review]
### General Feedback
- Thank you for the patience on my delay reviewing this PR :raised_hands: 
- Snippets looked formatted correctly and seem to do what's described

### Changes
- I left a comment with the proposed change, essentially the idea is to keep the snippets as "opt in" as we can. 

If you feel like it's the wrong way to handle this, feel free to leave a comment with your reasoning!

:sunglasses:

File: package.json (hunk 1 above)
Comment (wesleyegberto) [reply package.json:161]: Agree, this opt-in is a great feature, I will update to `java-testing`.

Thank you for the review =)

Comment (OkelleyDevelopment) [reply package.json]: Awesome! Thanks again :sunglasses:

**Pull request merged**